qiskit/transpiler/passes/optimization/commutative_cancellation.py

```diff
@@ -156,6 +156,10 @@ def run(self, dag):
                         current_angle = np.pi / 4
                     elif current_node.name == "s":
                         current_angle = np.pi / 2
+                    else:
+                        raise TranspilerError(
+                            f"Angle for operation {current_node.name } is not defined"
+                        )
 
```

Comment:
Do we either want a test for this, or to suppress coverage?

Reply:
done via https://github.com/Qiskit/qiskit/pull/12564